nixos/modules/services/display-managers/lemurs.nix

```
@@ -0,0 +1,118 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.displayManager.lemurs;
+  settingsFormat = pkgs.formats.toml { };
+in
+{
+  options.services.displayManager.lemurs = {
+    enable = lib.mkEnableOption "lemurs, a customizable TUI display/login manager";
+
+    package = lib.mkPackageOption pkgs "lemurs" { };
+
+    settings = lib.mkOption {
+      type = settingsFormat.type;
+      default = { };
+      example = lib.literalExpression ''
+        {
+          do_log = true;
+        }
+      '';
+      description = ''
+        Configuration for lemurs, provided as a Nix attribute set and automatically
+        serialized to TOML.
+        See [lemurs configuration documentation](https://github.com/coastalwhite/lemurs/blob/main/extra/config.toml) for available options.
+      '';
+    };
+
+    vt = lib.mkOption {
+      type = lib.types.ints.positive;
+      default = 2;
+      description = ''
+        The virtual console (tty) that lemurs should use.
+      '';
+    };
+  };
+  config = lib.mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = !config.services.displayManager.autoLogin.enable;
+        message = ''
+          lemurs doesn't support auto login.
+        '';
+      }
+    ];
+
+    security.pam.services.lemurs = {
+      unixAuth = true;
+      startSession = true;
+      # See https://github.com/coastalwhite/lemurs/issues/166
+      setLoginUid = false;
+      enableGnomeKeyring = lib.mkDefault config.services.gnome.gnome-keyring.enable;
+    };
+
+    environment.systemPackages = [ cfg.package ];
+
+    services = {
+      dbus.packages = [ cfg.package ];
+      seatd.enable = true;
+      xserver = {
+        # To enable user switching, allow lemurs to allocate TTYs/displays dynamically.
+        tty = null;
+        display = null;
+      };
+      displayManager = {
+        enable = true;
+        execCmd = "exec ${lib.getExe cfg.package} --config ${settingsFormat.generate "config.toml" cfg.settings}";
+        # set default settings
+        lemurs.settings =
+          let
+            sessionData = config.services.displayManager.sessionData.desktops;
```

nit: the name for this variable is confusing.
```suggestion
            desktops = config.services.displayManager.sessionData.desktops;
```